### PR TITLE
Wastebin LXC

### PR DESCRIPTION
**Pull request by MickLesk**
## Description

Wastebin is an personal Pastebin
https://github.com/matze/wastebin
We work with it a lot in our company and I have turned it into an LXC. 

## Type of change

Please delete options that are not relevant.

- [x] New Script
- [x] This change requires a documentation update

Project reworked a bit. Rust is a pig sometimes :D 
Do you think it's okay like this?
Old Pull-Request: 
https://github.com/tteck/Proxmox/pull/2605

### CODE REVIEW

**Comment by MickLesk** [comment]
Install:
image


Update:
image

File: ct/wastebin.sh [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+if (( $(df /boot | awk 'NR==2{gsub("%","",$5); print $5}') > 80 )); then
+  read -r -p "Warning: Storage is dangerously low, continue anyway? <y/N> " prompt
+  [[ ${prompt,,} =~ ^(y|yes)$ ]] || exit
+fi
+msg_info "Stopping Wastebin"
+systemctl stop wastebin
+msg_ok "Wastebin Stopped"
+
+msg_info "Updating Wastebin"
+RELEASE=$(curl -s https://api.github.com/repos/matze/wastebin/releases/latest | grep "tag_name" | awk '{print substr($2, 2, length($2)-3) }') &>/dev/null
```
Comment (tteck) [inline ct/wastebin.sh]: why `&>/dev/null`?

File: ct/wastebin.sh [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+cd /opt
+if [ -d wastebin_bak ]; then
+  rm -rf wastebin_bak
+fi
+mv wastebin wastebin_bak
+wget -q "https://github.com/matze/wastebin/archive/refs/tags/${RELEASE}.zip"
+unzip -q ${RELEASE}.zip &>/dev/null
```
Comment (tteck) [inline ct/wastebin.sh]: Why `&>/dev/null`? It has `-q`

File: install/wastebin-install.sh [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+color
+verb_ip6
+catch_errors
+setting_up_container
+network_check
+update_os
+
+msg_info "Installing Dependencies (Patience)"
+$STD apt-get install -y --no-install-recommends \
+  unzip \
```
Comment (tteck) [inline install/wastebin-install.sh]: `unzip` is a `mc` dependency

File: install/wastebin-install.sh [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+  curl \
+  sudo \
+  git \
+  make \
+  mc
+msg_ok "Installed Dependencies"
+
+msg_info "Installing Rust (Patience)" 
+$STD bash <(curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs) -y
+$STD source ~/.cargo/env
```
Comment (tteck) [inline install/wastebin-install.sh]: Doesn't need `$STD`

File: install/wastebin-install.sh [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+msg_ok "Installed Rust" 
+
+RELEASE=$(curl -s https://api.github.com/repos/matze/wastebin/releases/latest | grep "tag_name" | awk '{print substr($2, 2, length($2)-3) }')
+$STD wget -q "https://github.com/matze/wastebin/archive/refs/tags/${RELEASE}.zip"
```
Comment (tteck) [inline install/wastebin-install.sh]: Doesn't need `$STD`

File: install/wastebin-install.sh [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+$STD unzip -q ${RELEASE}.zip
```
Comment (tteck) [inline install/wastebin-install.sh]: Doesn't need `$STD`

File: install/wastebin-install.sh [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+mv wastebin-${RELEASE} /opt/wastebin
+rm -R ${RELEASE}.zip 
+cd /opt/wastebin
+$STD cargo build -q --release
```
Comment (tteck) [inline install/wastebin-install.sh]: Doesn't need `$STD`

File: install/wastebin-install.sh [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+
+[Service]
+User=root
+WorkingDirectory=/opt/wastebin
+ExecStart=/root/.cargo/bin/cargo run --release --quiet
+
+[Install]
+WantedBy=multi-user.target
+EOF
+systemctl daemon-reload
```
Comment (tteck) [inline install/wastebin-install.sh]: Doesn't need `systemctl daemon-reload`

**Comment by MickLesk** [comment]
Oh nice, thanks 4 the input. I fix it :-)

File: ct/wastebin.sh (hunk 1 above)
Comment (MickLesk) [reply ct/wastebin.sh]: Ah okay, I completely overlooked that. (y)

File: ct/wastebin.sh (hunk 2 above)
Comment (MickLesk) [reply ct/wastebin.sh]: you're right, it's a bad habit, in this case of course it's useless as it's done quietly anyway. I'll remember that for the next one :-)

File: install/wastebin-install.sh (hunk 3 above)
Comment (MickLesk) [reply install/wastebin-install.sh]: I didn't even know that! All right, thanks!

File: install/wastebin-install.sh (hunk 8 above)
Comment (MickLesk) [reply install/wastebin-install.sh]: Dont? I thought that had caused problems for me when I didn't have it in yet. But well, I trust you. :-)

**Review by MickLesk — COMMENTED** [review]
various fixes by @tteck comments

File: install/wastebin-install.sh [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+  build-essential \
+  curl \
+  sudo \
+  git \
+  make \
+  mc
+msg_ok "Installed Dependencies"
+
+msg_info "Installing Rust (Patience)" 
+bash <(curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs) -y
```
Comment (tteck) [inline install/wastebin-install.sh]: This needs a pre `$STD` or post `> /dev/null 2>&1`

**Comment by tteck** [comment]
Just curious, is your storage "docker" configured as LVM-Thin or is it simply a directory?

**Comment by MickLesk** [comment]
> Just curious, is your storage "docker" configured as LVM-Thin or is it simply a directory?

you can ignore the name, I have an Mini-PC (Intel Nuc) with 3 hard disks, 1 NVME, 1 SSD and an HDD. The HDD is for developing new LXC (was still lying around) and is named "docker" 

Its an simply LVM
image
image

File: install/wastebin-install.sh (hunk 9 above)
Comment (MickLesk) [reply install/wastebin-install.sh]: done

**Comment by MickLesk** [comment]
@tteck 
Little Fix:
The following was lost during merging
**msg_info "Installing Wastebin (Patience)"** 

-> wastebin-install.sh line 33

**Comment by tteck** [comment]
~Does that drive show under the LVM-Thin tab?~

Edit: Never-mind, you said it was a spinner.

**Pull request merged**